tests/queries/0_stateless/03174_projection_deduplicate.sql

```diff
@@ -0,0 +1,24 @@
+-- https://github.com/ClickHouse/ClickHouse/issues/65548
+DROP TABLE IF EXISTS test_projection_deduplicate;
+
+CREATE TABLE test_projection_deduplicate
+(
+    `id` Int32,
+    `string` String,
+    PROJECTION test_projection
+    (
+        SELECT id
+        GROUP BY id
+    )
+)
+ENGINE = MergeTree
+PRIMARY KEY id;
+
+INSERT INTO test_projection_deduplicate VALUES (1, 'one');
+INSERT INTO test_projection_deduplicate VALUES (1, 'one');
+
+OPTIMIZE TABLE test_projection_deduplicate DEDUPLICATE; -- { serverError NOT_IMPLEMENTED }
+
+SELECT * FROM test_projection_deduplicate;
```

Comment:
Could you add a line that checks OPTIMIZE TABLE ... after dropping the projection?